app/src/store/data.ts

```diff
@@ -349,5 +355,12 @@ export default function useDataStore() {
     grantRoundsCLRData: computed(() => grantRoundsCLRData.value),
     grantRoundMetadata: computed(() => grantRoundMetadata.value),
     grantRoundsDonationToken: computed(() => grantRoundsDonationToken.value),
+  } as DataStore;
+}
+
+// --- Singleton of the datastore ---
+export default function useDataStore() {
+  return {
+    ...(dataStore || (dataStore = getDataStore())),
```

Comment:
Yes thats correct. The methods inside what was `useDateStore` (now `getDataStore`) would be redefined, but now that I'm looking again - I don't think it matters. It's the `startPolling` fn which calls `rawInit` and that will only happen on a change of provider. This design still seems to make sense to me though because it will prevent us from redefining all of these methods - but I'm not convinced that its a crucial change.

Reply:
Just summarizing the discussion around this that we had during the standup yesterday:
- Right now we have `data.ts` which uses this store pattern, and the rest of the stores that use the original pattern
- If this pattern actually reduces loading time then we should either (1) switch all stores to it, or (2) add comments explaining why this one is different
- But I don't think this actually has an impact on loading times so I'd be in favor of reverting back to the original architecture (and this shouldn't have an impact, so if it does, there's a bug somewhere that should be fixed)